lib/ansible/module_utils/remote_management/dellemc/dellemc_idrac.py

```diff
@@ -0,0 +1,97 @@
+# _*_ coding: utf-8 _*_
+
+#
+# Dell EMC OpenManage Ansible Modules
+# Version 1.0
+# Copyright (C) 2018 Dell Inc.
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# All rights reserved. Dell, EMC, and other trademarks are trademarks of Dell Inc. or its subsidiaries.
+# Other trademarks may be trademarks of their respective owners.
+#
+
+from __future__ import (absolute_import, division,
+                        print_function)
+import tempfile
+import os
+__metaclass__ = type
+
+try:
+    from omsdk.sdkinfra import sdkinfra
+    from omsdk.sdkcreds import UserCredentials
+    from omsdk.sdkfile import FileOnShare, file_share_manager
+    from omsdk.sdkprotopref import ProtoPreference, ProtocolEnum
+    from omsdk.http.sdkwsmanbase import WsManOptions
+    HAS_OMSDK = True
+
+except ImportError:
+
+    HAS_OMSDK = False
+
+
+class iDRACConnection():
+    def __init__(self, module):
+        if HAS_OMSDK is False:
+            results = {}
+            results['msg'] = "Dell EMC OMSDK library is required for this module"
+            module.fail_json(**results)
```

Comment:
Please don't exit module execution from this layer, because it doesn't belong here: it's better to have all "exits" inside the end-module code.
Otherwise it's messy and requires you to keep track of the schema across multiple filed. Also, this looks like a reusable helpers and you will be conflicting with schemas of different modules where you'll use it.

The right way to do this is to raise an exception which preserves _semantics_ of what's happening (I think, `RuntimeError` is appropriate in this case). And then, catch it in the concrete module implementation returning appropriate value compliant with its schema.

I think, we need to work out the proper layering so that there'd be distinct difference between different logic levels, validation and returning the outcome.

Reply:
updated the code block